### PR TITLE
nvgpu one hot update

### PR DESCRIPTION
**Pull request by fengleitian**
1. handle float point data. 
2. add boundary check.

### CODE REVIEW

File: src/ngraph/runtime/gpu/cuda_emitter.cpp [hunk 1]
```diff
@@ -206,18 +207,19 @@ size_t runtime::gpu::CUDAEmitter::build_onehot(const std::array<std::string, 2>&
...
+        hot_axis_stride *= output_shape[i];
     }
-
+    uint32_t output_size = static_cast<uint32_t>(shape_size(output_shape) * output_datatype_size);
     // create the launch primitive
     std::unique_ptr<gpu::primitive> kernel_launch(
         new gpu::primitive{[=](void** inputs, void** outputs) mutable {
             std::vector<void*> args_list{
-                &inputs[0], &outputs[0], &repeat_size, &repeat_times, &nthreads};
+                &inputs[0], &outputs[0], &hot_axis_stride, &hot_axis_shape, &nthreads};
```
[the remark this answers is not in the document]
Comment (csullivan) [reply src/ngraph/runtime/gpu/cuda_emitter.cpp]: Oops, that's what I get for reviewing your PR during a meeting, this is the arg_list, _not_ the capture list. Sorry about that. Should be no problem here.

File: src/ngraph/runtime/gpu/cuda_emitter.cpp [hunk 2]
```diff
@@ -170,7 +170,8 @@ size_t runtime::gpu::CUDAEmitter::build_concat(const std::vector<std::string>& d
 size_t runtime::gpu::CUDAEmitter::build_onehot(const std::array<std::string, 2>& dtypes,
                                                NVShape input_shape,
                                                NVShape output_shape,
-                                               size_t one_hot_axis)
+                                               size_t one_hot_axis,
+                                               size_t output_datatype_size)
 {
     std::stringstream kernel_name;
     kernel_name << "onehot_" << join(dtypes, "_");
```
[the remark this answers is not in the document]
Comment (fengleitian) [reply src/ngraph/runtime/gpu/cuda_emitter.cpp]: yes.

**Comment by fengleitian** [comment]
@aprocter we will not provide exceptions on GPU side. and the implementation is different from spec now. Those test are been disabled.

**Comment by aprocter** [comment]
Works for me. I think we need to update the definition of `OneHot`. Not sure it was fully thought through at the time.

**Comment by fengleitian** [comment]
TESTNOW

**Pull request merged**